src/modules/iasql_functions/rpcs/iasql_get_sql_since.ts

```diff
@@ -0,0 +1,387 @@
+          .filter(([k, _]: [string, any]) => k !== 'ami' && k !== 'id');
+        const updatedEntries = Object.entries(cl.change?.change ?? {}).filter(
+          ([k, _]: [string, any]) => k !== 'id',
+        );
+        const updatedValues = await Promise.all(
+          updatedEntries.map(
+            async ([k, v]: [string, any]) =>
+              await findRelationOrReturnFormattedValue(cl.tableName, k, v, modsIndexedByTable, orm),
+          ),
+        );
+        const oldValues = await Promise.all(
+          originalEntries.map(
+            async ([k, v]: [string, any]) =>
+              await findRelationOrReturnFormattedValue(cl.tableName, k, v, modsIndexedByTable, orm),
+          ),
+        );
+        query = format(
+          `
+          UPDATE %I
+          SET ${updatedEntries
+            .map(
+              ([_, v]: [string, any]) =>
+                `${typeof v === 'object' && !Array.isArray(v) ? '%I::jsonb = %s' : '%I = %s'}`,
+            )
+            .join(', ')}
+          WHERE ${originalEntries
+            .map(
+              ([_, v]: [string, any]) =>
+                `${typeof v === 'object' && !Array.isArray(v) ? '%I::jsonb = %s' : '%I = %s'}`,
+            )
+            .join(' AND ')};
+        `,
+          cl.tableName,
+          ...updatedEntries.flatMap(([k, _]: [string, any], i) => [k, updatedValues[i]]),
+          ...originalEntries.flatMap(([k, _]: [string, any], i) => [k, oldValues[i]]),
+        );
+        break;
+      default:
+        break;
+    }
+    if (query) queries.push(query);
+  }
+  return queries;
+}
+
+/**
+ * @internal
+ * The changes from iasql_audit_log are stored as JSON.
+ * We need to look if the value is a relation to other table and return the respective sub-query
+ * or return a formatted value for the query.
+ */
+async function findRelationOrReturnFormattedValue(
+  tableName: string,
+  key: string,
+  value: any,
+  modsIndexedByTable: { [key: string]: ModuleInterface },
+  orm: TypeormWrapper,
+): Promise<string> {
+  // We might need to recreate a sub-query because it could be column referencing other table.
+  // For this we need to get Typeorm metadata for the `tableName` and inspect the columns and relations in order to recreate the sub-query if necessary.
+  // We need to recreate the sub-query because related columns might not be the same across databases connected to the same cloud account.
+  const metadata = await getMetadata(tableName, modsIndexedByTable, orm);
+  let columnMetadata: ColumnMetadata | undefined;
+  if (value && metadata) {
+    // If `metadata instanceof EntityMetadata` means that there's an Entity in Typeorm which it's table name is `tableName`
+    if (metadata instanceof EntityMetadata) {
+      columnMetadata = metadata.ownColumns
+        .filter(oc => oc.databaseName === key)
+        .map(oc => oc)
```

Comment:
What does this `.map` do?

Reply:
![Alt Text](https://media1.giphy.com/media/l36kU80xPf0ojG0Erg/giphy.gif?cid=ecf05e47kb7vgelcxu8kwfv8pe1co21aqenrtf21h88n7zm1&rid=giphy.gif&ct=g)

Reply:
Nothing, actually. Before I was getting the relation but then I decided to keep the column metadata for the final array check.